File: frontend/src/types/collection.ts
export interface Collection {
  id: string;
  name: string;
  description: string | null;
  category: string | null;
  itemCount: number;
  createdAt: string;
  updatedAt: string;
}

export interface CollectionItem {
  id: string;
  collectionId: string;
  scrapeJobId: string | null;
  data: Record<string, string | number | boolean | null>;
  isNew: boolean;
  scrapedAt: string;
}

File: backend/schemas/collection.py
"""Collection request/response schemas."""

from datetime import datetime

from pydantic import BaseModel, ConfigDict


class CollectionCreate(BaseModel):
    name: str
    description: str | None = None
    category: str | None = None


class CollectionOut(BaseModel):
    id: str
    name: str
    description: str | None
    category: str | None
    item_count: int
    created_at: datetime
    updated_at: datetime

    model_config = ConfigDict(from_attributes=True)


class CollectionItemOut(BaseModel):
    id: str
    collection_id: str
    scrape_job_id: str | None
    data: dict
    is_new: bool
    scraped_at: datetime

    model_config = ConfigDict(from_attributes=True)
